src/vtt-segment-loader.js

```diff
@@ -229,12 +221,17 @@ export default class VTTSegmentLoader extends SegmentLoader {
    */
   skipEmptySegments_(segmentInfo) {
     while (segmentInfo && segmentInfo.segment.empty) {
-      segmentInfo = this.generateSegmentInfo_(
-        segmentInfo.playlist,
-        segmentInfo.mediaIndex + 1,
-        segmentInfo.startOfSegment + segmentInfo.duration,
-        segmentInfo.isSyncRequest
-      );
+      // stop at the last possible segmentInfo
+      if (segmentInfo.mediaIndex + 1 >= segmentInfo.playlist.segments.length) {
```

Comment:
Looks like codecov is saying both these changes aren't covered by tests? Do we have any tests for this piece?